### PR TITLE
[TSK-237] move integ task to buildSrc

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts
Comment (adsukharev) [inline buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts:1]: com.qpay.java.tests.gradle.kts

File: buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_21
+}
+
+configurations {
+    compileOnly {
+        extendsFrom(configurations.annotationProcessor.get())
+    }
+}
```
Comment (adsukharev) [inline buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts]: configurations and java are not relevant for tests

File: buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+tasks.check { dependsOn(integrationTest) }
+
+dependencies {
+    integrationTestImplementation("org.junit.jupiter:junit-jupiter-api:${Versions.JUNIT_JUPITER_API}")
+    integrationTestImplementation("org.wiremock:wiremock-standalone:${Versions.WIREMOCK}")
+    integrationTestImplementation("org.springframework.boot:spring-boot-starter-test")
+    integrationTestRuntimeOnly("com.h2database:h2:${Versions.H2}")
```
Comment (adsukharev) [inline buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts]: This is a specific dep for projects which use relation db. It cannot be common for all

File: buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+idea {
+    module {
+        testSources.from(sourceSets[integrationTaskName].java.srcDirs)
+    }
+}
+
+tasks.check { dependsOn(integrationTest) }
+
+dependencies {
+    integrationTestImplementation("org.junit.jupiter:junit-jupiter-api:${Versions.JUNIT_JUPITER_API}")
```
Comment (adsukharev) [inline buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts]: junit is included in starter-test

File: buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    integrationTestImplementation("org.wiremock:wiremock-standalone:${Versions.WIREMOCK}")
+    integrationTestImplementation("org.springframework.boot:spring-boot-starter-test")
+    integrationTestRuntimeOnly("com.h2database:h2:${Versions.H2}")
+    integrationTestRuntimeOnly("org.junit.platform:junit-platform-launcher")
```
Comment (adsukharev) [inline buildSrc/src/main/kotlin/com.qpay.java.integration-test.gradle.kts]: junit is included in starter-test

File: payment-manager/build.gradle.kts [hunk 5]
```diff
@@ -29,9 +30,6 @@ dependencies {
     compileOnly("org.projectlombok:lombok")
 
     testImplementation("org.springframework.boot:spring-boot-starter-test")
```
Comment (adsukharev) [inline payment-manager/build.gradle.kts]: You have it in your custom plugin

**Review by adsukharev — COMMENTED** [review]
You only made it for 2 projects. But what about others?

**Comment by Hoocki** [comment]
> You only made it for 2 projects. But what about others?

others doesn't have any integration tests, but if it's necessary to create package 'integrationTest' i will add

**Comment by adsukharev** [comment]
> > You only made it for 2 projects. But what about others?
> 
> 
> 
> others doesn't have any integration tests, but if it's necessary to create package 'integrationTest' i will add

It's better to keep consistency across all projects

File: payment-manager/build.gradle.kts [hunk 6]
```diff
@@ -14,6 +15,10 @@ configurations {
     }
 }
 
+val integrationTestImplementation by configurations.getting {
+    extendsFrom(configurations.implementation.get())
+}
```
Comment (adsukharev) [inline payment-manager/build.gradle.kts]: I see, if we exlude some dependencies we have to add these lines to use. Not cool.
For simplicity, let's put all tests dependencies in your custom plugin.

Alternatively, if you have a big project you can create different plugins for sql, no sql integration tests. But I think right now, we can keep everything in one common plugin.

**Review by adsukharev — COMMENTED** [review]
Remove tasks.withType<Test> from all submodules

**Review by adsukharev — APPROVED** [review]

**Review by kostya4201337 — APPROVED** [review]

**Pull request merged**